From fe3ddc22e24a6815481c16e17cd1050bd91b5fe2 Mon Sep 17 00:00:00 2001
From: Michael Stahl <mstahl@redhat.com>
Date: Wed, 1 Feb 2012 15:49:16 +0100
Subject: [PATCH 2/4] fdo#37024: SwView::SwView: fix BROWSE_MODE setting:

Apparently there is a check in SwDoc::SetAllUniqueFlyNames so that when
frames with certain anchor types are in a document, then SetLoaded()
is never called.  Checking this flag via IsLoaded() here seems
unnecessary.
This problem was introduced with ebc5777548dea42ed966a16c66d879b1485bbfb4,
from CWS swlayoutrefactoring.

Signed-off-by: Noel Power <noel.power@novell.com>
---
 sw/source/ui/uiview/view.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sw/source/ui/uiview/view.cxx b/sw/source/ui/uiview/view.cxx
index f6912af..b2c920a 100644
--- a/sw/source/ui/uiview/view.cxx
+++ b/sw/source/ui/uiview/view.cxx
@@ -835,7 +835,7 @@ SwView::SwView( SfxViewFrame *_pFrame, SfxViewShell* pOldSh )
 
         if( !bOldShellWasSrcView && pWebDShell && !bOldShellWasPagePreView )
             aUsrPref.setBrowseMode( sal_True );
-        else if( rDoc.IsLoaded() )
+        else
             aUsrPref.setBrowseMode( rDoc.get(IDocumentSettingAccess::BROWSE_MODE) );
 
         //Fuer den BrowseMode wollen wir keinen Factor uebernehmen.
-- 
1.7.7.6

